### DISCUSSION TITLE
How to use on Ubuntu 22.04 ?

### DISCUSSION THREAD

**Discussion by ithakaa** [lldap/lldap#765] [General]
Is there a guide so we can use LLDAP on Ubuntu 22.04 ?

**Comment by nitnelave** [comment]
I'm not sure I understand the question: the readme has instructions on how to run it in docker or directly from source. What are you trying to do, and what are you having trouble with?

**Reply by ithakaa** [reply]
I'm using the package built and made available [here](https://software.opensuse.org//download.html?project=home%3AMasgalor%3ALLDAP&package=lldap)

however when setting the admin username and password in /etc/lldap/lldap_config.toml i still can't connect to the web interface

**Reply by martadinata666** [reply]
So I'm trying to replicate but I can't, this is my lldap_config.toml
```
## Tune the logging to be more verbose by setting this to be true.
## You can set it with the LLDAP_VERBOSE environment variable.
# verbose=false

## The host address that the LDAP server will be bound to.
## To enable IPv6 support, simply switch "ldap_host" to "::":
## To only allow connections from localhost (if you want to restrict to local self-hosted services),
## change it to "127.0.0.1" ("::1" in case of IPv6)".
#ldap_host = "0.0.0.0"

## The port on which to have the LDAP server.
#ldap_port = 3980

## The host address that the HTTP server will be bound to.
## To enable IPv6 support, simply switch "http_host" to "::".
## To only allow connections from localhost (if you want to restrict to local self-hosted services),
## change it to "127.0.0.1" ("::1" in case of IPv6)".
#http_host = "0.0.0.0"

## The port on which to have the HTTP server, for user login and
## administration.
#http_port = 17170

## The public URL of the server, for password reset links.
#http_url = "http://localhost"

## Random secret for JWT signature.
## This secret should be random, and should be shared with application
## servers that need to consume the JWTs.
## Changing this secret will invalidate all user sessions and require
## them to re-login.
## You should probably set it through the LLDAP_JWT_SECRET environment
## variable from a secret ".env" file.
## This can also be set from a file's contents by specifying the file path
## in the LLDAP_JWT_SECRET_FILE environment variable
## You can generate it with (on linux):
## LC_ALL=C tr -dc 'A-Za-z0-9!#%&'\''()*+,-./:;<=>?@[\]^_{|}~' </dev/urandom | head -c 32; echo ''
jwt_secret = "somejwtsecret"

## Base DN for LDAP.
## This is usually your domain name, and is used as a
## namespace for your users. The choice is arbitrary, but will be needed
## to configure the LDAP integration with other services.
## The sample value is for "example.com", but you can extend it with as
## many "dc" as you want, and you don't actually need to own the domain
## name.
#ldap_base_dn = "dc=example,dc=com"

## Admin username.
## For the LDAP interface, a value of "admin" here will create the LDAP
## user "cn=admin,ou=people,dc=example,dc=com" (with the base DN above).
## For the administration interface, this is the username.
ldap_user_dn = "admin"

## Admin email.
## Email for the admin account. It is only used when initially creating
## the admin user, and can safely be omitted.
#ldap_user_email = "admin@example.com"

## Admin password.
## Password for the admin account, both for the LDAP bind and for the
## administration interface. It is only used when initially creating
## the admin user.
## It should be minimum 8 characters long.
## You can set it with the LLDAP_LDAP_USER_PASS environment variable.
## This can also be set from a file's contents by specifying the file path
## in the LLDAP_LDAP_USER_PASS_FILE environment variable
## Note: you can create another admin user for user administration, this
## is just the default one.
ldap_user_pass = "adminadmin"

## Database URL.
## This encodes the type of database (SQlite, MySQL, or PostgreSQL)
## , the path, the user, password, and sometimes the mode (when
## relevant).
## Note: SQlite should come with "?mode=rwc" to create the DB
## if not present.
## Example URLs:
##  - "postgres://postgres-user:password@postgres-server/my-database"
##  - "mysql://mysql-user:password@mysql-server/my-database"
##
## This can be overridden with the LLDAP_DATABASE_URL env variable.
database_url = "sqlite:///var/lib/lldap/users.db?mode=rwc"

## Private key file.
## Contains the secret private key used to store the passwords safely.
## Note that even with a database dump and the private key, an attacker
## would still have to perform an (expensive) brute force attack to find
## each password.
## Randomly generated on first run if it doesn't exist.
## Alternatively, you can use key_seed to override this instead of relying on
## a file.
## Env variable: LLDAP_KEY_FILE
key_file = "/var/lib/lldap/private_key"

## Seed to generate the server private key, see key_file above.
## This can be any random string, the recommendation is that it's at least 12
## characters long.
## Env variable: LLDAP_KEY_SEED
#key_seed = "RanD0m STR1ng"

## Ignored attributes.
## Some services will request attributes that are not present in LLDAP. When it
## is the case, LLDAP will warn about the attribute being unknown. If you want
## to ignore the attribute and the service works without, you can add it to this
## list to silence the warning.
#ignored_user_attributes = [ "sAMAccountName" ]
#ignored_group_attributes = [ "mail", "userPrincipalName" ]

## Options to configure SMTP parameters, to send password reset emails.
## To set these options from environment variables, use the following format
## (example with "password"): LLDAP_SMTP_OPTIONS__PASSWORD
[smtp_options]
## Whether to enabled password reset via email, from LLDAP.
#enable_password_reset=true
## The SMTP server.
#server="smtp.gmail.com"
## The SMTP port.
#port=587
## How the connection is encrypted, either "NONE" (no encryption), "TLS" or "STARTTLS".
#smtp_encryption = "TLS"
## The SMTP user, usually your email address.
#user="sender@gmail.com"
## The SMTP password.
#password="password"
## The header field, optional: how the sender appears in the email. The first
## is a free-form name, followed by an email between <>.
#from="LLDAP Admin <sender@gmail.com>"
## Same for reply-to, optional.
#reply_to="Do not reply <noreply@localhost>"

## Options to configure LDAPS.
## To set these options from environment variables, use the following format
## (example with "port"): LLDAP_LDAPS_OPTIONS__PORT
[ldaps_options]
## Whether to enable LDAPS.
#enabled=true
## Port on which to listen.
#port=6360
## Certificate file.
#cert_file="/data/cert.pem"
## Certificate key file.
#key_file="/data/key.pem"

```

**Reply by ithakaa** [reply]
are you saying you can log into the web interface with this config ?

**Reply by martadinata666** [reply]
tldr; yes

**Reply by ithakaa** [reply]
that exact same config doesn;t work for me, I am unable to log into the web interface

did you install the pkg file manually ?

**Reply by martadinata666** [reply]
I had some sus, how about try login with
username: `admin`
password: `password`

**Reply by ithakaa** [reply]
ok that worked but why ?

**Reply by martadinata666** [reply]
I see, the package when installed `autostart` lldap by default thus `default` --> `admin` `password` created rather than modified `admin` `password` combo. Previously I got success because I stopped `lldap service`, wiped out the data at `/var/lib/lldap`, and start clean.

@Masgalor is it possible to not `autostart` lldap service by default?

**Reply by Masgalor** [reply]
@martadinata666 sorry for the late replay, I will look into it.

**Reply by Masgalor** [reply]
The build-recipes were updated to not start the service after installation.
The behavior of the systemd-units during package install/uninstall/upgrade is now more consistent between all distributions.
 - Service is not started if it is a new installation
 - Service is restarted after an upgrade if the service was running before (otherwise stays stopped)
 - Service is restarted after a forced reinstall if the service was running before (otherwise stays stopped)

I also found a minor permission problem with CentOS during testing, that's also fixed.

I could install lldap, change the admin-name and admin-password in the config, start the service and use my custom credentials to login. That was the problem here if I understood you correctly.

**Reply by martadinata666** [reply]
Ah yes, tried it. So far it works like expectes.
Need to update Readme.md again.
1. Install lldap
2. Set admin password and jwt
3. Enable systemd service.

if not set, default admin/password

**Comment by ithakaa** [comment]
Issue Resolved

**Comment by nitnelave** [comment]
Honestly, I was thinking about it, the simplest might be to just tell
people that the admin user/pass is "admin/password", and they can change it
in the UI after starting. That would get rid of a lot of confusion around
"I changed the password in the config, but I can't log in, why not??"

On Thu, Dec 21, 2023 at 1:47 PM Dedy Martadinata S ***@***.***>
wrote:

> Ah yes, tried it. So far it works like expectes.
> Need to update Readme.md again.
>
>    1. Install lldap
>    2. Set admin password and jwt
>    3. Enable systemd service.
>
> if not set, default admin/password
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lldap/lldap/discussions/765#discussioncomment-7918006>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGCPWK6ZAQSFLXWK6VQRPDYKQVUPAVCNFSM6AAAAABAX4LA42VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TSMJYGAYDM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Valentin Tolmer

**Comment by martadinata666** [comment]
I'm thinking of giving options/choices to deploy.
1. Install, and enable systems then we get `admin/password` default behaviour.
2. Or edit the configuration file if you need advanced modification before enabling the service.
